app/javascript/crayons/MarkdownToolbar/MarkdownToolbar.jsx

```diff
@@ -181,12 +182,40 @@ export const MarkdownToolbar = ({ textAreaId }) => {
   const insertSyntax = (syntaxName) => {
     setOverflowMenuOpen(false);
 
-    const { newTextAreaValue, newCursorStart, newCursorEnd } =
-      markdownSyntaxFormatters[syntaxName].getFormatting(textArea);
+    const {
+      newCursorStart,
+      newCursorEnd,
+      editSelectionStart,
+      editSelectionEnd,
+      replaceSelectionWith,
+    } = markdownSyntaxFormatters[syntaxName].getFormatting(textArea);
+
+    // We try to update the textArea with document.execCommand, which requires the contentEditable attribute to be true.
+    // The value is later toggled back to 'false'
+    textArea.contentEditable = 'true';
+    textArea.focus({ preventScroll: true });
+    textArea.setSelectionRange(editSelectionStart, editSelectionEnd);
+
+    try {
+      // We first try to use execCommand which allows the change to be correctly added to the undo queue.
+      // document.execCommand is deprecated, but the API which will eventually replace it is still incoming (https://w3c.github.io/input-events/)
+      if (replaceSelectionWith === '') {
+        document.execCommand('delete', false);
```

Comment:
I thought initially we could just do same as below, e.g.

```jsx
document.execCommand('insertText', false, '');
```

but unfortunately some browsers ignore an `insertText` when passed an empty string